src/generic_rosenbrock.jl

```diff
@@ -0,0 +1,62 @@
+struct RosenbrockTableau{T,T2}
+    a::Array{T,2}
+    C::Array{T,2}
+    b::Array{T,1}
+    gamma::T2
+    d::Array{T,1}
+    c::Array{T2,1}
+end
+
+macro _gen_constant_cache(inds,name,T)
+    quote
+        for ind in $inds
+            push!(cacheExpr.args,:($(Symbol($name,"$(ind[1])$(ind[2])"))=convert($T,$(tab.$name[ind]))))
+        end
+    end
+end
+
+function gen_constant_cache(tab::RosenbrockTableau,cacheName::Symbol)
+    ainds=findall(!iszero,tab.a)
+    Cinds=findall(!iszero,tab.C)
+    cacheExpr=quote end
+    @_gen_constant_cache ainds a :T
+    @_gen_constant_cache Cinds C :T
+    @_gen_constant_cache eachindex(tab.b) b :T
+    push!(cacheExpr.args,:(gamma=convert(T2,$(tab.gamma))))
+    @_gen_constant_cache eachindex(tab.d) d :T
+    @_gen_constant_cache eachindex(tab.c) c :T2
+
```

Comment:
This functions returns the last Expr that you pushed and I am not sure about why cacheExpr is needed.

Reply:
I'm working on this function. Eventually, it will return cacheExpr which is used inside the macro.